src/libostree/ostree-async-progress.c

```diff
@@ -424,6 +424,50 @@ ostree_async_progress_set_uint64 (OstreeAsyncProgress       *self,
   ostree_async_progress_set_variant (self, key, g_variant_new_uint64 (value));
 }
 
+static void
+copy_progress_state (gpointer key,
+                     gpointer value_pointer,
+                     gpointer dest_pointer)
+{
+  OstreeAsyncProgress *dest = OSTREE_ASYNC_PROGRESS (dest_pointer);
+  GVariant *value = NULL;
+
+  if (value_pointer)
+    {
+      value = g_variant_ref (value_pointer);
+      g_hash_table_replace (dest->values, key, value);
+    }
+}
+
+/**
+ * ostree_async_progress_copy_state:
+ * @self: An #OstreeAsyncProgress to copy from
+ * @dest: An #OstreeAsyncProgress to copy to
+ *
+ * Atomically copies all the state from @self to @dest, without invoking the
+ * callback.
+ * This is used for proxying progress objects across different GMainContexts.
+ *
+ * Since: 2019.3
```

Comment:
We’re on 2019.6 now, I think.

Reply:
Not until we rebase it some more, we're currently on 2019.3 ... I am working on the upstream patch and this annotation will have a different version there.